docs/architecture-decisions/0007-result-message-spec.md

```diff
@@ -0,0 +1,110 @@
+all data required for the consuming applications to process them.
+
+## Decision
+
+We will use the following result message specification:
+
+Each SQS Message sent by the dspace-submission-service to one of its result
+queues will contain two components, MessageAttributes and MessageBody.
+
+### MessageAttributes
+
+MessageAttributes is a JSON object containing two items, PackageID and
+SubmissionSource, structured like so:
+
+```json
+MessageAttributes = {
+    "PackageID": {
+        "DataType": "String",
+        "StringValue": "<A unique ID created by the submitting application that
+            will allow said application to match the result information to each
+            submitted package, e.g. 'etd_123123' or '98765'. The PackageID of
+            the result message must match the PackageID from the corresponding
+            submit message.>"
+    },
+    "SubmissionSource": {
+        "DataType": "String",
+        "StringValue": "<Name of the submitting system, e.g. 'ETD'. The
+          SubmissionSource of the result message must match the
+          SubmissionSource from the corresponding submit message.>"
+    }
+}
+```
+
+### MessageBody
+
+The MessageBody will contain detailed information about the submission result.
+The specification is slightly different for success results and error results.
+
+Note: SQS requires that the MessageBody be a string. However, this service and
+the submitting applications will want to parse/create the MessageBody as JSON
+objects, so the specification here shows the JSON object structure. This
+service must convert the MessageBody to a string before sending
+the message and the consuming applications must parse it back into a JSON
+object when processing messages from the result queue.
+
+#### Success Result
+
+```json
+MessageBody = {
+    "ResultType": "success",
+    "ItemHandle": "Handle of the successfully created new item in DSpace, e.g.
+        '1721.1/131022'",
+    "lastModified": "Timestamp the item was last modified in DSpace, e.g. 'Thu
+        Sep 09 17:56:39 UTC 2021'",
+    "Bitstreams": [
+        {
+            "BitstreamName": "Name of the Bitstream in DSpace, e.g.
+                'baker_report.pdf'",
+            "BitstreamUUID": "UUID of the Bitstream in DSpace",
+            "BitstreamChecksum": "JSON object with the value and hash algorithm
+                of the DSpace-calculated checksum as supplied by the bitstream
+                post, e.g.
+                {
+                    'value': 'a4e0f4930dfaff904fa3c6c85b0b8ecc',
+                    'checkSumAlgorithm': 'MD5'
+                }"
+        },
+        {
+            "(Repeat for all bitstreams posted with item.)"
+        }
+    ]
+}
+```
+
+#### Error Result
+
+```json
+MessageBody = {
+    "ResultType": "error",
+    "ErrorInfo": "Information provided by the DSpace Submission Service about
+        where in the process the error occurred, e.g. 'Error occurred while
```

Comment:
Similar to "lastModified" for success, it may be useful to have error related timestamp info (when the attempt was made to ingest content) for tracking/diagnostic purposes. This info could be useful to Wiley's business stakeholders, as opposed to tech stakeholders, who will have access to relevant logging presumably.

Reply:
Yeah, I agree this would be useful. I'll update the spec and make a ticket to add this to the DSS code. Thanks!